test/validation/api/ipsec/ipsec.c

```diff
@@ -1115,16 +1115,19 @@ int ipsec_init(odp_instance_t *inst, odp_ipsec_op_mode_t mode)
 		fprintf(stderr, "Packet pool creation failed.\n");
 		return -1;
 	}
-	if (mode == ODP_IPSEC_OP_MODE_ASYNC ||
-	    mode == ODP_IPSEC_OP_MODE_INLINE) {
+
+	if (suite_context.inbound_op_mode == ODP_IPSEC_OP_MODE_ASYNC ||
+	    suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_ASYNC ||
+	    suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_INLINE) {
```

Comment:
I am still not convinced about the usefulness of this patch. To me, the code gets more hairy, as instead of a function argument that tells the function what to do, dependencies to global variables get created. And subsequent patches remove much of what got changed by this patch.

The condition here has not been fixed. If inbound op mode is inline and outbound op mode is sync, this piece of code fails to create the destination queue (not the default queue as I said in the older comment). But maybe the easiest fix is to remove this patch?